qhub/stages/input_vars.py

```diff
@@ -152,6 +155,10 @@ def stage_04_kubernetes_ingress(stage_outputs, config):
         "certificate-secret-name": config["certificate"]["secret_name"]
         if config["certificate"]["type"] == "existing"
         else None,
+        "load-balancer-annotations": ingress_terraform_overrides.get(
```

Comment:
Using what I showed above we just supply `terraform_overrides` e.g. `**config.get("ingress", {}).get("terraform_overrides": {})`. Terraform variables will do all the checking for us.

Reply:
The problem is, we need `load-balancer-ip` and `load-balancer-annotations` separately, to be passed to different variables, so that they can be used at their designated places:

In `qhub/template/stages/04-kubernetes-ingress/modules/kubernetes/ingress/main.tf`

spec:

```hcl
  spec {
    selector = merge({
      "app.kubernetes.io/component" = "traefik-ingress"
    }, var.load-balancer-annotations)

```

and 

```hcl
    type = "LoadBalancer"
    load_balancer_ip = var.load-balancer-ip
```

Reply:
I think this should work. Suppose

```yaml
ingress:
  terraform_overrides:
      load-balancer-annotations:
           k1: v1
           k2: v2
      load-balancer-ip: 1.2.3.4
```

Then in `input-vars.py` we supply:

```
    return {
        "name": config["project_name"],
        "environment": config["namespace"],
        "node_groups": _calculate_note_groups(config),
        "enable-certificates": (config["certificate"]["type"] == "lets-encrypt"),
        "acme-email": config["certificate"].get("acme_email"),
        "acme-server": config["certificate"].get("acme_server"),
        "certificate-secret-name": config["certificate"]["secret_name"]
        if config["certificate"]["type"] == "existing"
        else None,
        **config.get("ingress", {}).get("terraform_overrides": {})
    }

```

This way `terraform_overrides` key can override any setting. Does this make sense?

Reply:
> This way `terraform_overrides` key can override any setting.

Yes, any variable present in the terraform. This makes more sense. I'll add this.